airbytehq/airbyte
# airbyte-ci: use connectors-qa instead of connector_ops.qa_check

## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/2905

This PR replaces the use of `connector_ops.qa_checks.py` for the new `connectors_qa` package (https://github.com/airbytehq/airbyte/pull/35322) in the `QAChecks` step which run on `airbyte-ci connectors test`

## How
* Call `connectors_qa` via a `SimpleDockerStep`
* Remove redundant step which are already checks in `connectors_qa`
<img width="1649" alt="Screenshot 2024-02-15 at 16 28 41" src="https://github.com/airbytehq/airbyte/assets/5551758/c2c81e58-a64f-40ef-b568-74b192834dfa">



## Manual testing
* Manually run `airbyte-ci connectors --name=source-faker test` to make sure no regression is introduced by the deletion of some steps. [Ran](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/manual/augustin_02-12-airbyte-ci_use_connectors-qa_instead_of_connector_opsqa_check/1708095830/aa30a14e5ed3216fd793d38a545bc488fb0beaed/source-amazon-ads/4.0.3/output.html) on `source-amazon-ads` which is 🟢 on `master`
* Manually run a pre-release of a connector for extra safety
